core/primitives/src/types.rs

```diff
@@ -116,20 +131,173 @@ pub enum StateChangeCause {
     ValidatorAccountsUpdate,
 }
 
+pub type RawStateChangesList = Vec<(StateChangeCause, Option<Vec<u8>>)>;
+
 /// key that was updated -> list of updates with the corresponding indexing event.
-pub type StateChanges =
-    std::collections::BTreeMap<Vec<u8>, Vec<(StateChangeCause, Option<Vec<u8>>)>>;
+pub type RawStateChanges = std::collections::BTreeMap<Vec<u8>, RawStateChangesList>;
 
-#[derive(Deserialize)]
+#[derive(Debug, Serialize, Deserialize)]
 #[serde(tag = "changes_type", rename_all = "snake_case")]
 pub enum StateChangesRequest {
-    AccountChanges { account_id: AccountId },
-    DataChanges { account_id: AccountId, key_prefix: Vec<u8> },
-    SingleAccessKeyChanges { account_id: AccountId, access_key_pk: PublicKey },
-    AllAccessKeyChanges { account_id: AccountId },
-    CodeChanges { account_id: AccountId },
-    SinglePostponedReceiptChanges { account_id: AccountId, data_id: CryptoHash },
-    AllPostponedReceiptChanges { account_id: AccountId },
+    AccountChanges {
+        account_id: AccountId,
+    },
+    SingleAccessKeyChanges {
+        account_id: AccountId,
+        access_key_pk: PublicKey,
+    },
+    AllAccessKeyChanges {
+        account_id: AccountId,
+    },
+    CodeChanges {
+        account_id: AccountId,
+    },
+    DataChanges {
+        account_id: AccountId,
+        #[serde(rename = "key_prefix_base64", with = "base64_format")]
+        key_prefix: StoreKey,
+    },
+}
+
+#[derive(Debug)]
+pub enum StateChangeValue {
+    AccountUpdate { account_id: AccountId, account: Account },
```

Comment:
Why `AccountUpdate` has `account_id` in it, but `AccessKeyUpdate` doesn't have `account_id`?